src/box/lua/config/instance_config.lua

```diff
@@ -1179,6 +1183,57 @@ return schema.new('instance_config', schema.record({
             end
         end,
     }),
+    feedback = schema.record({
+        crashinfo = schema.scalar({
+            type = 'boolean',
+            box_cfg = 'feedback_crashinfo',
+            default = true,
+            apply_default_if = feedback_apply_default_if,
+        }),
+        host = schema.scalar({
+            type = 'string',
+            box_cfg = 'feedback_host',
+            default = 'https://feedback.tarantool.io',
+            apply_default_if = feedback_apply_default_if,
+        }),
+        metrics_collect_interval = schema.scalar({
+            type = 'number',
+            box_cfg = 'feedback_metrics_collect_interval',
+            default = 60,
+            apply_default_if = feedback_apply_default_if,
+        }),
+        send_metrics = schema.scalar({
+            type = 'boolean',
+            box_cfg = 'feedback_send_metrics',
+            default = true,
+            apply_default_if = feedback_apply_default_if,
+        }),
+        enabled = schema.scalar({
+            type = 'boolean',
+            box_cfg = 'feedback_enabled',
+            default = true,
+            apply_default_if = feedback_apply_default_if,
+        }),
+        interval = schema.scalar({
+            type = 'number',
+            box_cfg = 'feedback_interval',
+            default = 3600,
+            apply_default_if = feedback_apply_default_if,
+        }),
+        metrics_limit = schema.scalar({
+            type = 'number',
+            box_cfg = 'feedback_metrics_limit',
+            default = 1024 * 1024,
+            apply_default_if = feedback_apply_default_if,
+        }),
+    }, {
+        validate = function(data, w)
+            if data == nil or box.internal.feedback_daemon ~= nil then
+                return
+            end
+            w.error('Feedback is disabled')
+        end,
+    }),
```

Comment:
AFAIU, it discards the following config on tarantool that has the feedback daemon build-time disabled.

```yaml
feedback: {}
```

However, I see nothing criminal in the config: neither of the feedback daemon options are set.

I propose to sink the validation function to individual parameters that should be discarded on the given instance with the given build type.

Reply:
BTW, I would allow the following config as well.

```yaml
feedback:
  enabled: false
```

The interesting question is what to do with the following one?

```yaml
feedback:
  enabled: false
  host: 'https://example.org'
```

Sending the feedback is not requested by this configuration, so we can accept it.

For example, strip all the feedback parameters before passing to `box.cfg()` in `applier/box_cfg.lua` if `feedback.enabled` is `false`.

OTOH, this is non-typical build type (we don't deliver such builds on our own) and it looks as not worthful give the effort.

So, I think that this comment can be ignored (but the one above looks simple and I would implement it).

Reply:
Moved validate to individual parameters. However, I think it's better to disable any values for the feedback parameters, since this logic looks simpler and clearer.